spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/adapters/loadbalancer/InstanceIsolationServiceInstanceFilter.java

```diff
@@ -86,7 +96,11 @@ public List<ServiceInstance> filter(ServiceInstanceListSupplier supplier, List<S
       }
 
       synchronized (lock) {
-        isolatedInstances.remove(InstanceIDAdapter.instanceId(serviceInstance));
+        if (checkInstanceHealth(serviceInstance)) {
```

Comment:
do not put check instance health in synchronized block

Reply:
fixed